bodhi/tests/server/test_utils.py

```diff
@@ -70,3 +72,50 @@ def test_sorted_builds(self):
         b1, b2 = sorted_builds([new, old])
         assert b1 == new, b1
         assert b2 == old, b2
+
+
+class TestCMDFunctions(unittest.TestCase):
+    @mock.patch('subprocess.Popen')
+    def test_stderr(self, mock_popen):
+        """
+        Ensures proper behavior when there is err output and the exit code isn't 0.
+        See https://github.com/fedora-infra/bodhi/issues/1412
+        """
+        mock_popen.return_value = mock.Mock()
+        mock_popen_obj = mock_popen.return_value
+        mock_popen_obj.communicate.return_value = ('output', 'error')
+        mock_popen_obj.returncode = 1
+        with mock.patch('bodhi.server.log.error') as mock_error:
+            cmd('/bin/echo', '"Hello World"')
+            self.assertTrue(mock_popen.called)
```

Comment:
If you're curious, I figured out what was causing the ValueError. The order of my decorators was the problem. I assumed the order in which you placed the patch was the order in which you added the parameter.
E.g.)

````
@mock.patch('some.thing.a')
@mock.patch('some.thing.b')
@mock.patch('some.thing.c')
def test_alphabet(a, b, c)
```
But it's really 
````
@mock.patch('some.thing.c')
@mock.patch('some.thing.b')
@mock.patch('some.thing.a')
def test_alphabet(a, b, c)
```

Well, now I know. :)